src/adr/0040-generalize-event-bus.md

```diff
@@ -0,0 +1,30 @@
+#40. Generalize EventBus
+
+Date: 2020-06-11
+
+## Status
+
+Proposed
+
+## Context
+
+When searching for users by quota ratio if we set the value of the parameters to 0, the search feature is supposed to return newly created users
+who have not received any email yet at that point. However, this is not the case because the quotas are currently being initialized only after
+a user receive the first email.
+
+We need to initialize user quotas upon user creation time. The problem is: there is currently no event at user creation 
+and since the quota-search feature is a plugin of James, it can not be hardwired into the domain logic of user management to initialize the quota for a just created user.
```

Comment:
```suggestion
and since the quota-search feature is a plugin of James, it cannot be hardwired into the domain logic of user management to initialize the quota for a just created user.
```